src/fastcs/transport/epics/gui.py

```diff
@@ -188,3 +200,38 @@ def extract_api_components(self, controller_api: ControllerAPI) -> Tree:
             components.append(Group(name=name, layout=Grid(), children=children))
 
         return components
+
+
+class PvaEpicsGUI(EpicsGUI):
+    """For creating gui in the PVA EPICS transport."""
+
+    def _get_pv(self, attr_path: list[str], name: str):
+        return f"pva://{super()._get_pv(attr_path, name)}"
+
+    def _get_read_widget(self, fastcs_datatype: DataType) -> ReadWidgetUnion | None:
+        if isinstance(fastcs_datatype, Table):
+            fastcs_datatypes = [
+                numpy_to_fastcs_datatype(datatype)
+                for _, datatype in fastcs_datatype.structured_dtype
+            ]
+
+            base_get_read_widget = super()._get_read_widget
+            widgets = [base_get_read_widget(datatype) for datatype in fastcs_datatypes]
+
+            return TableRead(widgets=widgets)  # type: ignore
+        else:
+            return super()._get_read_widget(fastcs_datatype)
+
+    def _get_write_widget(self, fastcs_datatype: DataType) -> WriteWidgetUnion | None:
+        if isinstance(fastcs_datatype, Table):
+            widgets = []
+            for _, datatype in fastcs_datatype.structured_dtype:
+                fastcs_datatype = numpy_to_fastcs_datatype(datatype)
+                widget = super()._get_write_widget(fastcs_datatype)
+                if isinstance(widget, ToggleButton):
+                    # Replace with compact version for Table row
+                    widget = CheckBox()
```

Comment:
Just a thought. Is this nicer?
```suggestion
                if isinstance(fastcs_datatype, Bool):
                    # Use compact check box in tables
                    widget = CheckBox()
                else:
                    widget = super()._get_write_widget(fastcs_datatype)
```